src/Adapter/MSTest.TestAdapter/Execution/TestExecutionManager.cs

```diff
@@ -314,37 +315,72 @@ private void ExecuteTestsInSource(IEnumerable<TestCase> tests, IRunContext? runC
                 }
 
                 var tasks = new List<Task>();
+                bool isSTA = false;
 
                 for (int i = 0; i < parallelWorkers; i++)
                 {
-                    tasks.Add(_taskFactory(() =>
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                        && MSTestSettings.RunConfigurationSettings.ExecutionApartmentState == ApartmentState.STA)
```

Comment:
Probably better not to check at every iteration. You could save the result as part of the `isSTA` variable you have created earlier.

Reply:
I made it partially with the isSTA as I have to check the OS before using STA or i get a warning